src/scheduler.cc

```diff
@@ -910,6 +910,12 @@ void Scheduler::tick(Locker& locker, int64 now) {
       // The process is already suppossed to preempt.
       // Check whether it is stuck.
       if (us_since_preemption <= MAX_RUN_WITHOUT_PREEMPTION_US) continue;
+      if (process->current_primitive_index() != -1) {
+        FATAL("Potential dead-lock detected in process %d; primitive call %d-%d\n",
```

Comment:
Let's make this consistent with https://github.com/toitlang/toit/blob/master/lib/core/exceptions.toit#L191.

```suggestion
        FATAL("Potential dead-lock detected in process %d; primitive call %d:%d\n",
```

Reply:
done.